File: app/src/main/java/ru/art2000/calculator/view_model/calculator/HistoryViewModel.java
package ru.art2000.calculator.view_model.calculator;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;

import java.util.List;

import ru.art2000.calculator.model.calculator.history.HistoryDatabaseItem;
import ru.art2000.calculator.model.calculator.history.HistoryListItem;

public interface HistoryViewModel {

    int COPY_ALL = 100;
    int COPY_EXPR = 101;
    int COPY_RES = 102;

    int DELETE = 200;

    LiveData<List<HistoryListItem>> getHistoryListItems();

    String copyHistoryItemToClipboard(@NonNull HistoryDatabaseItem item, int type);

    void removeHistoryItem(int id);

    void clearHistoryDatabase();

}

File: app/src/main/java/ru/art2000/calculator/view_model/calculator/HistoryViewModel.kt
package ru.art2000.calculator.view_model.calculator

import androidx.lifecycle.LiveData
import ru.art2000.calculator.model.calculator.history.HistoryListItem
import ru.art2000.calculator.model.calculator.history.HistoryDatabaseItem

interface HistoryViewModel {

    val historyListItems: LiveData<List<HistoryListItem>>

    fun copyHistoryItemToClipboard(item: HistoryDatabaseItem, type: Int): String

    fun updateHistoryItem(item: HistoryDatabaseItem)

    fun removeHistoryItem(item: HistoryDatabaseItem)

    fun clearHistoryDatabase()

    companion object {
        const val COPY_ALL = 100
        const val COPY_EXPR = 101
        const val COPY_RES = 102
        const val DELETE = 200
        const val COMMENT = 300
    }
}